### PR TITLE
Add log warning if SC detects if any ConfigSection's are being used

### PR DESCRIPTION
**Pull request by WilliamBZA**
Fixes https://github.com/Particular/ServiceControl/issues/1317

### CODE REVIEW

**Comment by WilliamBZA** [comment]
Does an XML check to avoid having to have the assembly in the dir to be able to load the configSection

File: src/ServiceControl/ServiceControl.csproj [hunk 1]
```diff
@@ -10,6 +10,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\ServiceControl.Transports.Msmq\ServiceControl.Transports.Msmq.csproj" />
```
Comment (SzymonPobiega) [inline src/ServiceControl/ServiceControl.csproj]: is this intentional?

Comment (WilliamBZA) [reply src/ServiceControl/ServiceControl.csproj]: I could do it as a separate PR, but it fixes the F5 experience

Comment (SzymonPobiega) [reply src/ServiceControl/ServiceControl.csproj]: Ok. I have no problem with this change as long as does not affect the packaging etc.

Comment (timbussmann) [reply src/ServiceControl/ServiceControl.csproj]: I don't know anything about SC packaging, but this change causes the msmq dlls to appear in the bin folder.

Comment (WilliamBZA) [reply src/ServiceControl/ServiceControl.csproj]: Hmm, I thought the other transports were removed during packaging, but that's only in the acceptance tests. I'll revert this commit for now and come up with a batter way to solve this

**Comment by WilliamBZA** [comment]
Reverted and rebased

File: src/ServiceControl/Infrastructure/Settings/InvalidConfigurationCheckFeature.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    logging.Warn("The use of AzureQueueConfig within ServiceControl has been deprecated. A Transport Adapater (https://docs.particular.net/servicecontrol/transport-adapter/) can be used if there are transport defaults that are not suitable for this environment.");
+                }
+            }
+
+            protected override Task OnStop(IMessageSession session)
+            {
+                return Task.CompletedTask;
+            }
+
+            ILog logging = LogManager.GetLogger(typeof(InvalidConfigurationCheck));
```
Comment (timbussmann) [inline src/ServiceControl/Infrastructure/Settings/InvalidConfigurationCheckFeature.cs]: not a big deal. but I wonder about the naming. Is this invalid or rather a deprecated configuration?

Comment (WilliamBZA) [reply src/ServiceControl/Infrastructure/Settings/InvalidConfigurationCheckFeature.cs]: I can get on board with a better name. I'll rename

File: src/ServiceControl/Infrastructure/Settings/DeprecatedConfigurationCheckFeature.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            EnableByDefault();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context) => context.RegisterStartupTask(b => b.Build<DeprecatedConfigurationCheck>());
+
+        class DeprecatedConfigurationCheck : FeatureStartupTask
+        {
+            protected override Task OnStart(IMessageSession session)
+            {
+                var config = XDocument.Load(AppDomain.CurrentDomain.SetupInformation.ConfigurationFile);
```
Comment (timbussmann) [inline src/ServiceControl/Infrastructure/Settings/DeprecatedConfigurationCheckFeature.cs]: is there any chance for that code to throw an exception we need to catch?
e.g. can the configuration file path be null or empty?

Comment (WilliamBZA) [reply src/ServiceControl/Infrastructure/Settings/DeprecatedConfigurationCheckFeature.cs]: If there is no app.config file, ServiceControl won't start as there are config settings that are required. If all of the requried settings reside in the machine.config, then this code will return Task.CompletedTask and the user won't get warned about the config sections. [Docs for this property](https://msdn.microsoft.com/en-us/library/system.appdomainsetup.configurationfile.aspx)

If this code can't load the location specified by this property, it means the account that ServiceControl is running under doesn't have access to that disk location, in which case it shouldn't start anyway - at the very least, the codepath wouldn't actually get this far anyway.

So I don't know if there's any specific checks we need to do here, but I suspect not.

**Pull request merged**